account_tax_balance/models/account_tax.py

```diff
@@ -0,0 +1,211 @@
+# © 2016 Lorenzo Battistini - Agile Business Group
+# © 2016 Antonio Espinosa <antonio.espinosa@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+
+
+class AccountTax(models.Model):
+    _inherit = 'account.tax'
+
+    balance = fields.Float(
+        string="Total Balance", compute="_compute_balance",
+    )
+    base_balance = fields.Float(
+        string="Total Base Balance", compute="_compute_balance",
+    )
+    balance_regular = fields.Float(
+        string="Balance", compute="_compute_balance",
+    )
+    base_balance_regular = fields.Float(
+        string="Base Balance", compute="_compute_balance",
+    )
+    balance_refund = fields.Float(
+        string="Balance Refund", compute="_compute_balance",
+    )
+    base_balance_refund = fields.Float(
+        string="Base Balance Refund", compute="_compute_balance",
+    )
+    has_moves = fields.Boolean(
+        string="Has balance in period",
+        compute="_compute_has_moves",
+        search="_search_has_moves",
+    )
+
+    def get_context_values(self):
+        context = self.env.context
+        return (
+            context.get('from_date', fields.Date.context_today(self)),
+            context.get('to_date', fields.Date.context_today(self)),
+            context.get('company_id', self.env.user.company_id.id),
+            context.get('target_move', 'posted'),
+        )
+
+    def _account_tax_ids_with_moves(self):
+        """ Return all account.tax ids for which there is at least
+        one account.move.line in the context period
+        for the user company.
+
+        Caveat: this ignores record rules and ACL but it is good
+        enough for filtering taxes with activity during the period.
+        """
+        req = """
+            SELECT id
+            FROM account_tax at
+            WHERE
+            company_id = %s AND
+            EXISTS (
+              SELECT 1 FROM account_move_Line aml
+              WHERE
+                date >= %s AND
+                date <= %s AND
+                company_id = %s AND (
+                  tax_line_id = at.id OR
+                  EXISTS (
+                    SELECT 1 FROM account_move_line_account_tax_rel
+                    WHERE account_move_line_id = aml.id AND
+                      account_tax_id = at.id
+                  )
+                )
+            )
+        """
+        from_date, to_date, company_id, target_move = self.get_context_values()
+        self.env.cr.execute(
+            req, (company_id, from_date, to_date, company_id))
+        return [r[0] for r in self.env.cr.fetchall()]
+
+    @api.multi
+    def _compute_has_moves(self):
+        ids_with_moves = set(self._account_tax_ids_with_moves())
+        for tax in self:
+            tax.has_moves = tax.id in ids_with_moves
+
+    @api.model
+    def _search_has_moves(self, operator, value):
+        if operator != '=' or not value:
```

Comment:
please consider moving the `'='` to a separate method or constant for ease of extensibility